gbpservice/nfp/core/event.py

```diff
@@ -10,283 +10,339 @@
-    create internal events.
+        # Unique id of the event, use what user passed or
+        # generate a new unique id.
+        uuid = kwargs.get('key', pyuuid.uuid4())
+        uuid_id = kwargs.get('id')
+        if uuid_id and uuid:
+            uuid = str(uuid)+str(uuid_id)
+        self.uuid = uuid
+        # see 'Event Types'
+        self.type = kwargs.get('type')
+        # see 'Event Flag'
+        self.flag = kwargs.get('flag')
+        # PID of worker which is handling this event
+        self.worker = kwargs.get('worker')
+        # Polling descriptor of event
+        self.poll_desc = kwargs.get('poll_desc')
+
+    def from_desc(self, desc):
+        self.type = desc.type
+        self.flag = desc.flag
+        self.worker = desc.worker
+        self.poll_desc = desc.poll_desc
+
+    def to_dict(self):
+        return {'uuid': self.uuid,
+                'type': self.type,
+                'flag': self.flag,
+                'worker': self.worker,
+                'poll_desc': self.poll_desc
+        }
+
+"""Defines the event structure.
+
+    Nfp modules need to create object of the class
+    to create an event.
 """
 
 
 class Event(object):
 
     def __init__(self, **kwargs):
-        # ID of the event, can be same for multiple events
+        # ID of event as passed by module
         self.id = kwargs.get('id')
-        # Module context, not decoded by core
-        self.data = kwargs.get('data', None)
-        # Handler used only @the time of registration
-        self.handler = kwargs.get('handler', None)
-        # To serialize this event.
-        self.serialize = kwargs.get('serialize', False)
-        # Events with same binding_key are related.
-        self.binding_key = kwargs.get('binding_key', None)
-        # Lifetime of event in seconds
+        # Data blob
+        self.data = kwargs.get('data')
+        # Whether to sequence this event w.r.t
+        # other related events.
+        self.sequence = kwargs.get('serialize', False)
+        # Unique key to be associated with the event
+        self.key = kwargs.get('key')
+        # Binding key to define relation between
+        # different events.
+        self.binding_key = kwargs.get('binding_key')
+        # Handler of the event.
+        self.handler = kwargs.get('handler')
+        # Lifetime of the event in seconds.
         self.lifetime = kwargs.get('lifetime', 0)
-        # Max number of times this event can be polled.
-        # Default, till stopped or forever.
-        self.max_times = -1
         # Identifies whether event.data is zipped
         self.zipped = False
-        # Added for log metadata
-        self.context = kwargs.get('context', None)
+        # Log metadata context
+        self.context = kwargs.get('context', {})
+        # Prepare the base descriptor
+        if self.key:
+            desc = EventDesc(**{'key': self.key,
+                                'id': self.id})
```

Comment:
Why to create a dict first and then expand, why not pass args directly
EventDesc(key=self.key, id=self.id)